draft-ietf-tls-tls13.md

```diff
@@ -2204,23 +2214,19 @@ The "extension_data" field of this extension contains a
 
 %%% Key Exchange Messages
 
-       enum { psk_ke(0), psk_dhe_ke(1), (255) } PskKeyExchangeMode;
-       enum { psk_auth(0), psk_sign_auth(1), (255) } PskAuthenticationMode;
-
-       struct {
-           PskKeyExchangeMode ke_modes<1..255>;
-           PskAuthenticationMode auth_modes<1..255>;
-           opaque identity<0..2^16-1>;
-       } PskIdentity;
-
+       opaque PskIdentity<0..2^16-1>;
+       opaque PskBinderEntry<0..255>;
+       
        struct {
            select (Handshake.msg_type) {
                case client_hello:
-                   PskIdentity identities<6..2^16-1>;
-
+                  PskIdentity identities<6..2^16-1>;
+                  PskBinderEntry binders<0..2^16-1>;
```

Comment:
Is the zero length ever attainable given that the identities array must be nonempty and a binder is needed for each PSK offered?